spring-cloud-dataflow-module-deployers/spring-cloud-dataflow-module-deployer-kubernetes/src/main/java/org/springframework/cloud/dataflow/module/deployer/kubernetes/ContainerFactory.java

```diff
@@ -0,0 +1,16 @@
+package org.springframework.cloud.dataflow.module.deployer.kubernetes;
+
+import org.springframework.cloud.dataflow.core.ModuleDeploymentRequest;
+
+import io.fabric8.kubernetes.api.model.Container;
+
+/**
+ * Defines how a Kubernetes {@link Container} is created. 
+ * 
+ * @author Florian Rosenberg
+ */
+public interface ContainerFactory {
```

Comment:
I'm not sure we'd want to merge the `PreBaked..` strategy, so discussion about it should be in a separate PR